great_expectations/execution_engine/sparkdf_execution_engine.py

```diff
@@ -261,9 +260,6 @@ def get_batch_data_and_markers(
                     """
                 )
 
-        elif isinstance(batch_spec, GCSBatchSpec):
-            raise NotImplementedError("Currently unsupported.")
-
```

Comment:
Could you provide some context around this? Do we not need a batch spec here?

Reply:
@cdkini Great question -- remarkably, for all cases other than Azure, the fact that the given BatchSpec is a subclass of PathBatchSpec makes it work.  The Azure case is different, because Spark and Pandas access the storage using different protocol formats, but for Pandas, S3, and GCS, there is one access protocol format that they all support.  Having said that, there is a TODO for Azure as well (to try to bring it in: `https://superconductive.atlassian.net/browse/IPV-248`).

Reply:
Hmm very interesting! Would you mind leaving a comment to describe that? Can be super brief but I think future users might be surprised to see a lack of cloud-specific BatchSpecs

Reply:
@cdkini DocString added.  Thanks.